pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -0,0 +1,117 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the BasisStatePreparation template.
+"""
+
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class QutritBasisStatePreparation(Operation):
+    r"""
+    Prepares a basis state on the given wires using a sequence of TShift gates.
+
+    .. warning::
+
+        ``basis_state`` influences the circuit architecture and is therefore incompatible with
+        gradient computations.
+
+    Args:
+        basis_state (array): Input array of shape ``(n,)``, where n is the number of wires
+            the state preparation acts on.
+        wires (Iterable): wires that the template acts on
+
+    **Example**
+
+    .. code-block:: python
+
+        dev = qml.device("default.qutrit", wires=4)
+
+        @qml.qnode(dev)
+        def circuit(basis_state, obs):
+            qml.QutritBasisStatePreparation(basis_state, wires=range(4))
+            return [qml.expval(qml.THermitian(obs, wires=i)) for i in range(4)]
+
+        basis_state = [0, 1, 1, 0]
+        obs = np.array([[1, 1, 0], [1, -1, 0], [0, 0, np.sqrt(2)]]) / np.sqrt(2)
+
+    >>> print(circuit(basis_state, obs))
+    [array(0.70710678), array(-0.70710678), array(-0.70710678), array(0.70710678)]
+    """
+    num_params = 1
+    num_wires = AnyWires
+    grad_method = None
+
+    def __init__(self, basis_state, wires, do_queue=True, id=None):
+        basis_state = qml.math.stack(basis_state)
+
+        # check if the `basis_state` param is batched
+        batched = len(qml.math.shape(basis_state)) > 1
+
+        state_batch = basis_state if batched else [basis_state]
+
+        for i, state in enumerate(state_batch):
+            shape = qml.math.shape(state)
+
+            if len(shape) != 1:
+                raise ValueError(
+                    f"Basis states must be one-dimensional; state {i} has shape {shape}."
+                )
+
+            n_bits = shape[0]
+            if n_bits != len(wires):
+                raise ValueError(
+                    f"Basis states must be of length {len(wires)}; state {i} has length {n_bits}."
+                )
+
+            if any(bit not in [0, 1, 2] for bit in state):
+                raise ValueError(
+                    f"Basis states must only consist of 0s, 1s, and 2s; state {i} is {state}"
+                )
+
+        # TODO: basis_state should be a hyperparameter, not a trainable parameter.
+        # However, this breaks a test that ensures compatibility with batch_transform.
+        # The transform should be rewritten to support hyperparameters as well.
```

Comment:
If this will not be addressed in this PR, could you please open an issue for this so that the core team can track it and hopefully make changes to make this work?

Reply:
I have created the issue with the information I have: #4223

Reply:
I see actually this is issue with not having `state` as a hyper parameter is somewhat related to your comment regarding the assertion checking, as these operators don't have an __init__ on their own but only inherited from Operations. 

I found this on the `adding_operators.rst`:

```
*Input checks.* Checking the inputs of the operation introduces an overhead and clashes with tools like
  just-in-time compilation. Find a balance of adding meaningful sanity checks (such as for the shape of tensors),
  but keeping them to a minimum.
```

Could this explain why of the absence of these checks?